### PR TITLE
IBX-7968: Removed obsolete internal doc & added varnish6.vcl

### PR DESCRIPTION
**Pull request by dabrt**
| :ticket: Issue | IBX-7968 |
|----------------|-----------|

#### Description:

Community reports issues when using Varnish 6.6.
We need to cleatly state what version is supported.

- [x] Ask for Code Review.

### CODE REVIEW

**Review by vidarl — CHANGES_REQUESTED** [review]
varnish5.vcl is for varnish 6.0LTS ( not for varnish 7.1)
varnish7.vcl is for varnish 7.1

So varnish5.vcl should technically be renamed varnish6.vcl, but that might be considered a BC break, so maybe copy it?

File: docs/varnish/vcl/varnish5.vcl [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // Varnish VCL for:
-// - Varnish 6.0 or higher (6.0LTS recommended, and is what we mainly test against)
+// - Varnish 6.0LTS or 7.1 (6.0LTS recommended, and is what we mainly test against)
```
Comment (dabrt) [inline docs/varnish/vcl/varnish5.vcl:2]:
```suggestion
// - Varnish 6.0LTS
```

**Review by vidarl — APPROVED** [review]
In Ibexa DXP 5.0, `varnish5.vcl` should be removed, or at least made deprecated

File: docs/varnish/.DS_Store
Comment (mnocon) [inline docs/varnish/.DS_Store:1]: This file should be removed (I recommend adding the .DS_STORE files to the global .gitignore, we can discuss this next week)

**Review by mnocon — COMMENTED** [review]
I agree that we need to clear the confusion around supported versions, but I'm afraid that adding the varnish6 file and maintaining both of them (varnish5, varnish6) at the same time won't work - I feel that that eventually someone forgets to update one of them.

I'm not sure what the best solution is (if we can't rename the file then I'd probably just keep the filenames varnish5 and varnish6 and made it right in v5 🤔 )

File: docs/varnish/.DS_Store
Comment (dabrt) [reply docs/varnish/.DS_Store:1]: Done

**Comment by dabrt** [comment]
> I agree that we need to clear the confusion around supported versions, but I'm afraid that adding the varnish6 file and maintaining both of them (varnish5, varnish6) at the same time won't work - I feel that that eventually someone forgets to update one of them.

@alongosz We have confronting opinions from Vidar and Marek. What's your say here?

**Review by alongosz — COMMENTED** [review]
@dabrt @mnocon @vidarl
I have a suggestion. Let's make a symlink to varnish5.vcl instead. It looks weird on the github diff and I need to check how this behaves on windows, but this is seems what Symfony is doing as well to avoid duplicating recipes.

In a long term I think the place of these files should be in recipes and they should be placed in custom projects. But not sure where exactly and that's for 5.0

@dabrt one more thing, I see you've stated that you're targeting main version. Do you indeed intend to make this change for 5.0 only? main is for 5.0, yet from the context of the discussion I'm not sure if that was your intention

**Comment by alongosz** [comment]
> It looks weird on the github diff and I need to check how this behaves on windows

update: sadly it doesn't work on windows (at leastOOTB)... we need to think what to do with it.

**Comment by vidarl** [comment]
May idea was to do the renaming in 5.0 only at least.. But it might be that we can only deprecate it in 5.0, and to the actuall renaming in v6?

**Comment by alongosz** [comment]
> May idea was to do the renaming in 5.0 only at least.. But it might be that we can only deprecate it in 5.0, and to the actuall renaming in v6?

We should rather rename it in 5.0.

**Comment by alongosz** [comment]
@dabrt we've discussed this within PHP Team and due to issues with symlinks decided to keep both files as you've originally intended, but I've added an explanation note on the top of each file. The changes have been already applied to the PR.
// FYI @mnocon @vidarl 

@dabrt the only issue remains is that these changes now target future 5.0 branch (main) and I'm not sure if this was your intention. If it was then we just can remove varnish5.vcl.

**Comment by dabrt** [comment]
> @dabrt the only issue remains is that these changes now target future 5.0 branch (main) and I'm not sure if this was your intention. If it was then we just can remove varnish5.vcl.

Please advise which bundle branch I should clone these changes to, to include it in 4..6 as well.

**Review by alongosz — APPROVED** [review]

**Review by konradoboza — APPROVED** [review]

**Review by Nattfarinn — COMMENTED** [review]
Shouldn't it be rebranded by the way? There are few `sub ez_*` and `call ez_*`that could be renamed.

**Comment by alongosz** [comment]
> Shouldn't it be rebranded by the way? There are few `sub ez_*` and `call ez_*`that could be renamed.

If we can safely do it, we should. 
@vidarl are there any BC concerns here? E.g. is it possible for someone to rely on thouse subroutine names or are they internal to a vcl file?

**Comment by vidarl** [comment]
> If we can safely do it, we should. @vidarl are there any BC concerns here? E.g. is it possible for someone to rely on those subroutine names or are they internal to a vcl file?

Clients can do al kind of changes in the .vlc file, so strictly speaking, any change we do in vcl can potentially be a BC.
But it is fair to call those sub routines that are named `ez_*` as internal. I would not consider changing their name in 5.0 a problem. But maybe keep them as-is in existing releases

**Review by mnocon — APPROVED** [review]

**Comment by adamwojs** [comment]
@Nattfarinn We will handle rebranding as part of the 5.0 scope

**Review by adamwojs — APPROVED** [review]

**Review by alongosz — APPROVED** [review]
Thx for the input, Vidar :+1:

**Comment by alongosz** [comment]
Given this is a copy of existing and working VCL, no QA required here. Merging as-is. Thanks everyone for the input.

**Pull request merged**